### PR TITLE
boundary check

### PR DESCRIPTION
**Pull request by BetterWang**
error: array subscript [-Werror=array-bounds]
added limit checks to header code

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test
the error doesn't seem to be related

https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10947/7653/runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log
```
A fatal system signal has occurred: bus error
... one stack follows

A fatal system signal has occurred: segmentation violation
```

... maybe a glitch?

**Comment by Dr15Jones** [comment]
The crash log appears to be missing the stack trace for the thread that actually crashed. We had this trouble before but thought we had fixed it (the fix is in 7_6).
@bbockelm

File: RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h [hunk 1]
```diff
@@ -202,30 +202,30 @@ class HiEvtPlaneFlatten {
...
+  double getYDB(unsigned int bin)  const {return flatYDB_[bin];}
+
+
+  double getCnt(unsigned int bin)  const {return flatCnt_[bin];}
+  void setXDB(unsigned int indx, double val) {flatXDB_[indx]=val;}
+  void setYDB(unsigned int indx, double val) {flatYDB_[indx]=val;}
+  void setXoffDB(unsigned int indx, double val) {xoffDB_[indx]=val;}
+  void setYoffDB(unsigned int indx, double val) {yoffDB_[indx]=val;}
+  void setPtDB(unsigned int indx, double val) {ptDB_[indx]=val;}
+  void setPt2DB(unsigned int indx, double val) {pt2DB_[indx]=val;}
```
Comment (cvuosalo) [inline RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h]: You added "if(bin<MAXCUTOFF)" checks to getPtDB() and getPt2DB(). Why not add these checks to all these other methods, lines 205-228? Is there any reason not to?

Comment (BetterWang) [inline RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h]: Sorry for the late reply.
We could, however, the origin of the error message was not clear to me.
The index and bin are calculated from the confdb, which can basically make sure there is no over/under flow.
This fix is just to suppress the warning.

Comment (cvuosalo) [inline RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h]: OK, as long as there is protection against out-of-bounds errors, further changes aren't essential.

**Comment by cvuosalo** [comment]
+1

For #10947 9ab47959d3cd6848667d79256f9bf2c51d6274c1

Small fix to prevent array-boundary warnings from the compiler. No change in run-time behavior should occur.

The code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10947/7656/summary.html) against baseline CMSSW_7_6_X_2015-08-25-1100 show no significant differences, as expected.

**Comment by davidlange6** [comment]
+1

**Pull request merged**